src/rbd.cc

```diff
@@ -3034,6 +3064,28 @@ if (!set_conf_param(v, p1, p2, p3)) { \
   // the relevant parts
   set_pool_image_name(imgname, (char **)&poolname,
 		      (char **)&imgname, (char **)&snapname);
+
+  if(!is_valid_name(poolname) || !is_valid_name(snapname) || !is_valid_name(imgname)) {
```

Comment:
All these if statements should be a lot simpler.  Build an error string from the names "pool" "snap" "image" and depending on how many invalid items either separate with a "," and/or "and."